### DISCUSSION TITLE
How to bulk import markdown posts?

### DISCUSSION THREAD

**Discussion by eduf** [GetPublii/Publii#1709] [Help]
How to bulk import markdown posts into Publii? Is it possible to just put the files in a folder and Publii does the rest (as it is done by the majority of the static generators)?

**Comment by bjazmoore** [comment]
I know that Publii uses SQLite for some functionality.  I am not sure when that data is populated and updated in the process.  My thought is that it happens at post save.  It is quite likely that just putting some MD files into a directory will not allow you to see these in the posts list interface.

How many are you thinking your import?  If it were 15 or less I would just do it manually.  Markdown is already formatted by nature, so the process should not be too bad.  If it is more that 15 I would get a macro program (like AutoIT) and write a script to create a new markdown post, copy from source, paste into the page.  Give it a title and edit the date if needed.  Save and publish.  - Repeat.

**Comment by bjazmoore** [comment]
It occurs to me that there is one other choice.  You could programmatically convert your MD files so that they conformed to the Wordpress WXR import/export standard.  Publii can bulk import WXR files coming from Wordpress and presumably any CMS that can generate a valid WXR file.  Check this out in the help file: https://getpublii.com/docs/import-wordpress-into-static-html-site.html.  Here is a pretty good explanation of a WXR file: https://ipggi.wordpress.com/2011/03/16/the-wordpress-extended-rss-wxr-exportimport-xml-document-format-decoded-and-explained/

**Comment by eduf** [comment]
Ok. Thanks.
I think the fastest way to do it right now would be just importing the markdown files into a local WP installation. Then convert the result into Publii.

**Reply by candidexmedia** [reply]
Let us know how that goes!
